tests/internal/test_settings.py

```diff
@@ -107,8 +107,8 @@ def _deleted_rc_config():
             "env": {"DD_TRACE_HEADER_TAGS": "X-Header-Tag-1,X-Header-Tag-2,X-Header-Tag-3:specific_tag3"},
             "expected": {
                 "trace_http_header_tags": {
-                    "X-Header-Tag-1": "X-Header-Tag-1",
-                    "X-Header-Tag-2": "X-Header-Tag-2",
+                    "X-Header-Tag-1": "",
+                    "X-Header-Tag-2": "",
```

Comment:
I don't have the full context on this test scenario, but input like `DD_TRACE_HEADER_TAGS=X-Header-Tag-1` result in span tag like `http.request.headers.x-header-tag-1` (Unless for the test this env is used for, the request/response does not contain `X-Header-Tag-1` and `X-Header-Tag-2` headers?)

Reply:
This test only asserts how the header to tag mapping is stored. If the header maps to an empty string there is a helper function that sets the correct name: https://github.com/DataDog/dd-trace-py/blob/5ecd4164c981d57595f800565371d62e7704e8e1/ddtrace/contrib/trace_utils.py#L153. 

I'll create a test that better captures this logic.

Reply:
Here's the test: [f00a719](https://github.com/DataDog/dd-trace-py/pull/9589/commits/f00a719773bd10ea5b164914ba874ffee79151e6)